packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -114,6 +131,29 @@ export class PendingStateManager implements IDisposable {
 		return this.pendingMessages.length + this.initialMessages.length;
 	}
 
+	/**
+	 * The minimumPendingMessageSequenceNumber is the minimum of the first pending message and the first initial message.
+	 *
+	 * We need this so that we can properly keep local data and maintain the correct sequence window.
+	 */
+	public get minimumPendingMessageSequenceNumber(): number | undefined {
+		const firstInitialMessage = this.initialMessages.peekFront();
```

Comment:
I do not think we care about this.initialMessages. This queue represents already acked messages, and if we have anything in there, that means Container's referenceSequenceNumber is below sequence number of any of these messages (and thus MSN is also below). So they can be safely ignored. Besides that, Daniel may chance what properties these messages have -  see https://github.com/microsoft/FluidFramework/pull/20846

Reply:
No longer caring about first initial messages